### ISSUE TITLE
weechat

### ISSUE THREAD

**Issue by sachaz** [nylira/prism-break#1749]
"WeeChat does not offer encryption by default, but there is a FiSH encryption plugin available."

Fish is considered as broken. Better to talk about OTR plugin

**Comment by strugee** [comment]
> Fish is considered as broken

Citation Needed, but even so it still seems to me that this should be removed. I've never heard of FiSH before, so it can't have had that much scrutiny. Seems like it's mostly just a naive "run everything through Blowfish" protocol, AFAICT? I could be wrong though.

In addition to OTR having an advantage in the level of scrutiny, it also has an advantage in that your contacts are more likely to be familiar with it. Therefore, marking as accepting PRs.

**Comment by bezcitu** [comment]
@strugee 

Maybe this is the source of @sachaz?

> NOTE ABOUT DH1080:
=================
Diffie-Hellman key exchange assumes that you already have
authenticated channels between Alice and Bob.  Which means that Alice
has to be sure that she is really talking to Bob and not to any man in
the middle.  But since the whole idea of FiSH is that you want to
encrypt your communication on the IRC server whose operators you do
not trust, there is no reliable way for Alice to tell if she really is
talking to Bob.  It could also be some rogue IRC admin impersonating
Bob with a fake hostname and ident or even doing a MITM attack on
DH1080.  This means **you can consider using DH1080 key exchange over
IRC utterly broken in terms of security**.

https://weechat.org/scripts/source/fish.py.html/

**Comment by Atavic** [comment]
FiSH python is based on PyCrypto module.

>Fish is considered as broken.

Source? I can list many fish implementations:

https://github.com/freshprince/weechat-fish
https://github.com/falsovsky/FiSH-irssi
https://github.com/emigonza/XChat-FiSH-AES
https://github.com/flakes/mirc_fish_10

**Comment by sachaz** [comment]
Sorry for not sourcing, I was relating about @strugee reference on DH 1080

**Comment by Zegnat** [comment]
I feel that note is a little confusing in its writing. Is it saying that key exchange is broken because that does not happen P2P but through the untrusted IRC server? That is the case for many encryption systems though, I believe including OTR.

This is why, after a key exchange through untrusted channels, you should use a side-channel (a phone call is often good for this) to read out the fingerprints of the keys you have. That is how you confirm there is no MITM going on. Again: that applies to every single encryption system.

**Comment by strugee** [comment]
@Atavic you're right, sorry - I wrote that comment pretty late at night and jumped the gun

@Zegnat but it seems like it does a Diffie-Hellman exchange for each conversation right? That would mean you had to verify the exchange for each conversation, as opposed to OTR, which has the concept of a long-term identity key which you only have to verify once.

AFAICT FiSH also lacks a lot of stuff that OTR does, like ratcheting. It also _seems_ like it'd be vulnerable to things like replay attacks, but I can't actually tell (for all of these things) - is the protocol actually properly documented anywhere?

**Comment by Zegnat** [comment]
> <i>[…]</i> is the protocol actually properly documented anywhere?

Not that I could find.

[The best SECURITY document](https://github.com/flakes/mirc_fish_10/blob/master/SECURITY.md) I have been able to turn up comes from [flakes/mirc_fish_10](https://github.com/flakes/mirc_fish_10) as linked by @Atavic. They implement FiSH and have quiet a list of things not to do. The “[Using FiSH Securely](https://github.com/flakes/mirc_fish_10/blob/master/SECURITY.md#using-fish-securely)” section is especially interesting, and does make it sound like people are probably using it wrong.

* So it FiSH broken? Not really, though you should make sure not to use ECB mode. (When was that ever a good idea‽)
* Is FiSH hard to use? Yes.
* Is OTR objectively easier than FiSH? … I don’t know.
* Is OTR protecting you against more attack vectors? Looks like it.

**Comment by Atavic** [comment]
Kudos to @flakes for that, otherwise FiSH seemed like close to abadonware status.

**Comment by strugee** [comment]
@Zegnat thanks for the pointers. Looks like there's a long list of things OTR is better at - replay attack preventing, forward secrecy, etc. Honestly I see absolutely no reason why FiSH would be preferred over OTR other than " because I feel like it." Shall I mark as accepting PRs again?

**Comment by Zegnat** [comment]
Again, don’t take my word for gospel, I have not taken a deep look into FiSH.

The one thing that FiSH seems to support is encrypted group chats (i.e. encrypting an entire IRC channel). Group chats got some work with mpOTR but haven’t landed in libotr as far as I know. (Correct me if I am wrong, would love to be wrong!) OMEMO might be an answer but I don’t think anyone has that working over IRC.

We can change the recommendation to the weechat-otr plugin, but be careful as this really is only meant for 1-on-1 chats.

Overall, IRC is not where you want to do your secret conversations.

**Comment by Atavic** [comment]
FiSH for Weechat hasn't been developed for years, but [FiSH for irssi](https://github.com/falsovsky/FiSH-irssi) is active. So, it seems that the lack of support is limited to Weechat.

**Comment by Zegnat** [comment]
@Atavic note that development does not necessarily mean it is getting more and more secure. If FiSH functions [as described by flakes/mirc_fish_10](https://github.com/flakes/mirc_fish_10/blob/master/SECURITY.md) there may just be no way to get rid off the weaknesses, continued development or not.

**Comment by strugee** [comment]
Marking as accepting PRs given that this protocol's security properties are questionable at best and it's underdocumented to the point where we're not even able to assess it properly. Plus, OTR covers most of the use cases FiSH does anyway, and it covers them better.
